### PR TITLE
Progress Bar

### PR DESCRIPTION
**Pull request by faradawn**
Created the ProgressBar:
- Takes in 4 number props: 'mastered', 'familiar', 'remaining,' and 'bonus'.
- If 'bonus' === 0, doesn't display 'bunus'  
- Did my best to make responsive - prevent the texts from overlapping
- Is put in /components/ItemDisplay/ProgressBar

If there is anything I need to do, please let me know!

### CODE REVIEW

File: frontend/src/learning/components/ItemDisplay/ProgressBar.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
+import React, {Component} from 'react'
```
Comment (Miclin1024) [inline frontend/src/learning/components/ItemDisplay/ProgressBar.js]: Please remember the trailing semis, and indention using 4 spaces :)

File: frontend/src/learning/components/ItemDisplay/ProgressBar.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  padding: 10px 30px;
+`
+// Outer bar container
+const BarContainer = styled.div`
+  background-color: #d8d8d8;
+	position: relative;
```
Comment (Miclin1024) [inline frontend/src/learning/components/ItemDisplay/ProgressBar.js]: This line was indented with 1 tab, please use 4 spaces instead

File: frontend/src/learning/components/ItemDisplay/ProgressBar.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+  constructor(props){
+    super(props)
+
+    this.total = this.props.mastered + this.props.familiar + this.props.remaining + this.props.bonus
+
+    this.bar = {
+        bar1: {
+          width: `${this.props.mastered / this.total * 100}%`,
+          background: "rgb(85, 162, 30)",
```
Comment (Miclin1024) [inline frontend/src/learning/components/ItemDisplay/ProgressBar.js]: All strings should be quoted with single quotes, just like you did for the imports.

File: frontend/src/learning/components/ItemDisplay/ProgressBar.js (hunk 2 above)
Comment (chenyx512) [reply frontend/src/learning/components/ItemDisplay/ProgressBar.js]: @Faradawn you should set up ur IDE to do this automatically

File: frontend/src/learning/components/ItemDisplay/ProgressBar.js (hunk 1 above)
Comment (chenyx512) [reply frontend/src/learning/components/ItemDisplay/ProgressBar.js]: 4 space indention is crucial, and you should set it up in your IDE automatically

**Review by Miclin1024 — APPROVED** [review]
This looks really good! Other than a few styling errors that you might wanna look out for in the future @Faradawn, I think we are ready to merge.

**Comment by faradawn** [comment]
Hi. Thanks for pointing out the styling mistakes! Sorry was a little
hurried. Will make sure to ESlint next time!

Best,
Faradawn

**Pull request merged**